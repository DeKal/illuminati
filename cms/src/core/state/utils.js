export const toSuccess = actionName => `${actionName}Success`

export const toError = actionName => `${actionName}Error`

export const toRequest = actionName => `${actionName}Request`
