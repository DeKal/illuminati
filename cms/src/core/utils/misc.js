import { POSTS_PER_PAGE } from 'core/const/posts'

export const isPathMatchUrl = (pathName, url, isExact = false) => {
  if (url === '/' || pathName === '/' || isExact) {
    return pathName === url
  }
  return pathName.includes(url)
}

export const stringtifyTags = (tags, limits = 20) => {
  const tagsStr = tags.join(', ')
  if (tagsStr.length > limits) {
    return tagsStr.slice(0, limits) + '...'
  }
  return tagsStr
}

export const calcMaxRowsPerPage = postCount => {
  const postsPerPageOptions = []
  for (let index = 0; index < POSTS_PER_PAGE.length; index++) {
    postsPerPageOptions.push(POSTS_PER_PAGE[index])
    if (POSTS_PER_PAGE[index] > postCount) {
      break
    }
  }
  return postsPerPageOptions
}

export const arrayToAttributes = array =>
  array.reduce((object, attribute) => {
    return {
      ...object,
      [attribute]: true
    }
  }, {})

export const attributesToArray = obj =>
  Object.keys(obj).map(attribute => (obj[attribute] ? attribute : null))
