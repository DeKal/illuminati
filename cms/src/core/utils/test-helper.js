import React from 'react'
import { shallow, mount } from 'enzyme'
import Theme from 'core/theme/ThemeWrapper'

export const shallowWithTheme = component => {
  const wrapper = shallow(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]').shallow()
}

export const mountWithTheme = component => {
  const wrapper = mount(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]')
}
