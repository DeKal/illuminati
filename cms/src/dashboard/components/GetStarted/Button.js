import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'

const ItemButton = withStyles(theme => ({
  root: {
    fontSize: theme.fontSizes.M,
    backgroundColor: theme.colors.primary,
    marginTop: theme.spaces.gutter,
    padding: `${theme.spaces.xsmall}px ${theme.spaces.medium}px`,
    textTransform: 'none'
  }
}))(Button)

export default ItemButton
