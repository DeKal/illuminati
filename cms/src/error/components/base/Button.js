import { styled } from '@material-ui/core/styles'

const Button = styled('a')(({ theme }) => ({
  fontSize: theme.fontSizes.base,
  textDecoration: 'none',
  textTransform: 'uppercase',
  background: theme.colors.primary,
  display: 'inline-block',
  padding: '15px 30px',
  borderRadius: '40px',
  color: theme.colors.primaryBg,
  fontWeight: '700',
  boxShadow: `0 4px 15px -5px ${theme.colors.primary}`,
  border: 'none',
  margin: `${theme.spaces.xsmall}px`,
  cursor: 'pointer'
}))

export default Button
