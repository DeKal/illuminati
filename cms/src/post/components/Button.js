import { styled } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'

const PostButton = styled(Button)(({ theme }) => ({
  backgroundColor: theme.colors.primary,
  marginLeft: '8px',
  padding: '12px 24px'
}))

export default PostButton
