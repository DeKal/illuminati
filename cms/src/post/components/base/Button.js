import { styled } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'

const PostButton = styled(Button)(({ theme }) => ({
  backgroundColor: theme.colors.primary,
  marginLeft: `${theme.spaces.base}px`,
  padding: '12px 24px'
}))

export default PostButton
