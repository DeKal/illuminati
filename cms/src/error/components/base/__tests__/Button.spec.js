import * as React from 'react'
import { mountWithTheme } from 'core/utils/test-helper'
import Button from '../Button'

describe('Button', () => {
  it('should render correctly', function() {
    const wrap = mountWithTheme(<Button />)
    expect(wrap).toMatchSnapshot()
  })
})
