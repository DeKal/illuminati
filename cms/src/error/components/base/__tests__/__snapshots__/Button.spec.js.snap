// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <Styled(a)>
    <a
      className="a-root-1 a-root-2"
    />
  </Styled(a)>
</div>
`;
